api_specifications/note-api-0.1.0.yaml

```diff
@@ -0,0 +1,256 @@
+  version: 1.0.0
+tags:
+  - name: Note
+    description: APIs to create, update, search to a case note.
+  - name: Constitute Element
+    description: APIs related to search constitute element to a case type.
+paths:
+  /note/v1/create:
+    post:
+      tags:
+        - Note
+      summary: Create new note in the registry
+      requestBody:
+        description: Details for the new note + RequestInfo meta data.
+        content:
+          '*/*':
+            schema:
+              $ref: '#/components/schemas/NoteRequest'
+        required: true
+      responses:
+        '201':
+          description: RequestInfo with note details created successfully
+          content:
+            '*/*':
+              schema:
+                $ref: '#/components/schemas/NoteResponse'
+        '400':
+          description: Note creation failed
+          content:
+            '*/*':
+              schema:
+                $ref: https://raw.githubusercontent.com/egovernments/DIGIT-OSS/master/core-services/docs/common-contract_v1-1.yml#/components/schemas/ErrorRes
+  /note/v1/update:
+    post:
+      tags:
+        - Note
+      summary: Update/edit a note
+      description: API to update the note. Updates can include constitute element, note.
+      requestBody:
+        description: Details for updating all editable fields in the note + RequestInfo meta data.
+        content:
+          '*/*':
+            schema:
+              $ref: '#/components/schemas/NoteRequest'
+        required: true
+      responses:
+        '200':
+          description: ResponseInfo with note details updated successfully
+          content:
+            '*/*':
+              schema:
+                $ref: '#/components/schemas/NoteResponse'
+        '400':
+          description: note updation failed
+          content:
+            '*/*':
+              schema:
+                $ref: https://raw.githubusercontent.com/egovernments/DIGIT-OSS/master/core-services/docs/common-contract_v1-1.yml#/components/schemas/ErrorRes
+  /note/v1/search:
+    post:
+      tags:
+        - Note
+      summary: Search for notes
+      requestBody:
+        description: Search criteria + RequestInfo meta data.
+        content:
+          '*/*':
+            schema:
+              $ref: '#/components/schemas/NoteSearchRequest'
+        required: true
+      responses:
+        '200':
+          description: ResponseInfo with note matching search criteria
+          content:
+            '*/*':
+              schema:
+                $ref: '#/components/schemas/NoteListResponse'
+        '400':
+          description: Note search failed. Refer to response body for specifics.
+          content:
+            '*/*':
+              schema:
+                $ref: https://raw.githubusercontent.com/egovernments/DIGIT-OSS/master/core-services/docs/common-contract_v1-1.yml#/components/schemas/ErrorRes
+  /element/v1/search:
+    post:
```

Comment:
just element in the name will eb confusing.  make this 

/constituentelement/v1/create:
/constituentelement/v1/update:
/constituentelement/v1/search:

Reply:
ok